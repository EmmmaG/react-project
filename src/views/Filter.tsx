import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import DropdownSelect, { Props as DropdownSelectProps } from '../components/DropdownSelect'
import TextInput, { Props as TextInputProps } from '../components/TextInput'

const styles = (theme: Theme) => ({
	filterContainer: {
		backgroundColor: theme.palette.grey[100],
		marginTop: '30px',
		marginLeft: '35px',
		marginRight: '35px',
		paddingLeft: '20px',
		borderBottom: `2px solid ${theme.palette.grey[300]}`,
	},
	/* eslint-disable quote-props */
	applyButton: {
		color: 'white',
		backgroundColor: '#0ebfae',
		height: '35px',
		width: '100px',
		marginTop: '11px',
		textTransform: 'none' as 'none',
		fontWeight: 700,
		'&:hover': {
			background: '#0e978a',
		},
	},
	/* eslint-enable quote-props */
})

type DropdownSelectP = Pick<DropdownSelectProps, Exclude<keyof DropdownSelectProps, 'classes'>>
type TextInputP = Pick<TextInputProps, Exclude<keyof TextInputProps, 'classes'>>

export interface Props extends WithStyles<typeof styles> {
	keywords: TextInputP
    launchPad: DropdownSelectP
    minYear: DropdownSelectP
	maxYear: DropdownSelectP
	onFilterSearch: () => void
}

export class Filter extends React.Component<Props> {
private renderDropdownSelects = () => {
	const dropdownSelectData = [this.props.launchPad, this.props.minYear, this.props.maxYear]

	return dropdownSelectData.map(data => <DropdownSelect key={`${data.title}_${data.value}`} {...data} />)
}

private renderTextInput = () => <TextInput {...this.props.keywords} />

private renderApplyButton = () =>
	<Button
		className={this.props.classes.applyButton}
		onClick={this.props.onFilterSearch}
	>
		Apply
	</Button>

public render() {
	const { classes } = this.props

	return (
		<div className={classes.filterContainer} id='filter'>
			{this.renderTextInput()}
			{this.renderDropdownSelects()}
			{this.renderApplyButton()}
		</div>
	)
}
}

export default withStyles(styles)(Filter)
