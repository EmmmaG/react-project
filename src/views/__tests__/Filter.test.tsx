import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import Filter, { Props } from '../Filter'

const props: Props = {
	keywords: {
		title: 'Keywords',
		onChange: jest.fn(),
		value: 'keyyy',
	},
	launchPad: {
		title: 'Launch Pad',
		onChange: jest.fn(),
		value: 'launchPadValue',
		options: [{ value: 'launchPadValue', text: 'launchPadText'}],
		width: '220px',
	},
	minYear: {
		title: 'Min Year',
		onChange: jest.fn(),
		value: 'minyearValue',
		options: [{ value: 'minyearValue', text: 'minyearText'}],
		width: '110px',
	},
	maxYear: {
		title: 'Max Year',
		onChange: jest.fn(),
		value: 'maxyearValue',
		options: [{ value: 'maxyearValue', text: 'maxyearText'}],
		width: '110px',
	},
	onFilterSearch: jest.fn(),
	classes: idObj,
}

describe('Filter', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<Filter {...props} />)
		expect(wrapper).toMatchSnapshot()
	})
})
