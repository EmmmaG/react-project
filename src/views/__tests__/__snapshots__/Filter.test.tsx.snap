// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<Filter
  classes={
    Object {
      "applyButton": "Filter-applyButton-2",
      "filterContainer": "Filter-filterContainer-1",
    }
  }
  keywords={
    Object {
      "onChange": [MockFunction],
      "title": "Keywords",
      "value": "keyyy",
    }
  }
  launchPad={
    Object {
      "onChange": [MockFunction],
      "options": Array [
        Object {
          "text": "launchPadText",
          "value": "launchPadValue",
        },
      ],
      "title": "Launch Pad",
      "value": "launchPadValue",
      "width": "220px",
    }
  }
  maxYear={
    Object {
      "onChange": [MockFunction],
      "options": Array [
        Object {
          "text": "maxyearText",
          "value": "maxyearValue",
        },
      ],
      "title": "Max Year",
      "value": "maxyearValue",
      "width": "110px",
    }
  }
  minYear={
    Object {
      "onChange": [MockFunction],
      "options": Array [
        Object {
          "text": "minyearText",
          "value": "minyearValue",
        },
      ],
      "title": "Min Year",
      "value": "minyearValue",
      "width": "110px",
    }
  }
  onFilterSearch={[MockFunction]}
/>
`;
